### PR TITLE
Update crates

### PR DESCRIPTION
**Pull request by gowthamsk-arm**
This MR has the following changes:

- Bump parsec-related crates
- Cargo update changes

Signed-off-by: Gowtham Suresh Kumar <gowtham.sureshkumar@arm.com>

### CODE REVIEW

**Review by ionut-arm — APPROVED** [review]
👍🏻

**Comment by gowthamsk-arm** [comment]
The fuzz test failure has been fixed and should pass the next nightly. 
https://github.com/rust-lang/rust/issues/109208

File: e2e_tests/docker_image/parsec-service-test-all.Dockerfile [hunk 1]
```diff
@@ -9,17 +9,18 @@ FROM ubuntu:18.04
...
-RUN apt install -y python3-distutils libclang-6.0-dev protobuf-compiler python3-pip
+RUN apt install -y python3-distutils libclang-11-dev protobuf-compiler python3-pip
+RUN apt install -y libgcrypt20-dev uuid-dev
 RUN DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt-get -y install tzdata
 WORKDIR /tmp
 
-# Download and install TSS 2.0
-RUN git clone https://github.com/tpm2-software/tpm2-tss.git --branch 2.3.3
+# Download and install TSS 3.2.0
+RUN git clone https://github.com/tpm2-software/tpm2-tss.git --branch 3.2.0
```
[the remark this answers is not in the document]
Comment (gowthamsk-arm) [reply e2e_tests/docker_image/parsec-service-test-all.Dockerfile:23]: We also compile parsec 1.0.0 to generate keys for the sqlite and ondisk KIM. 1.0.0 uses an older version of tss-esapi crate aborts with an error if 4.0.1 is used.

[the remark this answers is not in the document]
Comment (anta5010) [reply e2e_tests/docker_image/parsec-service-test-all.Dockerfile:23]: can we build two versions of tpm2-tss - for parsec 1.0.0 and the current version of parsec? While using TSS v3 in CI we might miss issues similar to https://github.com/parallaxsecond/parsec/issues/663

[the remark this answers is not in the document]
Comment (gowthamsk-arm) [reply e2e_tests/docker_image/parsec-service-test-all.Dockerfile:23]: Will try adding both

**Pull request closed without merging**